import-automation/progress-dashboard-rest/test/test_base_database.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for base_database.py.
+
+The Datastore emulator must be running for these tests to run.
```

Comment:
Tests are run automatically without human intervention at arbitrary times.  For example, the system may choose to run all the tests once per day, alerting test owners upon a failure.  Thus, manual set-up cannot work for tests (or code).

I am not sure what the solution is.  It might be to add a test fixture (http://screen/SPfHnaBnqcg) for the emulator.  I hope there is a Python interface to the emulator.  If not https://medium.com/google-cloud/using-google-cloud-emulators-for-integration-tests-7812890ebe0d, https://github.com/googleapis/google-cloud-go/issues/592, or https://github.com/googleapis/nodejs-datastore/issues/11 might have some hints.  (Updated: I see you have a setup below.)

Reply:
will investigate

Reply:
found out a way